addons/website_sale/models/product_template.py

```diff
@@ -309,8 +309,13 @@ def _price_with_tax_computed(
         )
         show_tax_excluded = self.user_has_groups('account.group_show_line_subtotals_tax_excluded')
         tax_display = 'total_excluded' if show_tax_excluded else 'total_included'
-        # The list_price is always the price of one.
-        return taxes.compute_all(price, pricelist.currency_id, 1, product, partner)[tax_display]
+        return taxes.compute_all(
+            price_unit=price,
+            currency=pricelist.currency_id,
+            quantity=1,  # `list_price` is always the price of one
+            product=product.sudo(),  # tax computation may require access to restricted fields
```

Comment:
Little question to be sure, are all the fields read on the product in `compute_all` non relational fields (or whose comodels are not restricted by company) ?
If not, the sudo might cause the multi-company record rules to not be applied when it maybe should.

Reply:
> If not, the sudo might cause the multi-company record rules to not be applied when it maybe should.

Can you elaborate?

Reply:
The `account_tax_python` module allows you you define tax formulas on based on arbitrary product fields AFAICT, so if you have a problematic field in mind, I could give it a try.

Reply:
The only field I have in mind is the taxes one, which is already handled beforehand and filtered before calling `compute_taxes` on it.  I took a quick look and the `product` variable doesn't seem to be used for anything else in the standard code, so it should be ok like this.

What i meant was that if you read `product.some_model_ids` and there is a multi-company `ir.rule` to filter out records from other companies, it would be skipped because of the `sudo` and the returned value might hold some records from other companies.  But it's not allowed to access relational fields in a python tax computation, so it's fine too.